fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -311,7 +311,8 @@ private void deleteDocument(Tuple groupingKey, Integer partitionId, Tuple primar
 
     @Override
     public CompletableFuture<Void> mergeIndex() {
-        return directoryManager.mergeIndex(partitioner, indexAnalyzerSelector.provideIndexAnalyzer(""));
+        return partitioner.rebalancePartitions()
```

Comment:
I didn't realize this in the first pass, but this doesn't play well with `AgilityContext`, namely we will rebalance the partition, and then, the `AgilityContext` will create a new transaction to do merges, and, if this partition ever commits it will fail.
Let's leave this as is, and followup with a better transaction story.